packages/grafana-runtime/src/services/EchoSrv.ts

```diff
@@ -120,6 +123,13 @@ let singletonInstance: EchoSrv;
  */
 export function setEchoSrv(instance: EchoSrv) {
   singletonInstance = instance;
+
+  // Check if there were any events reported to the FakeEchoSrv (before the main EchoSrv was initialized and track them)
+  if (instance.events) {
+    for (const event of instance.events) {
+      instance.addEvent(event as Omit<EchoEvent, 'meta'>);
+    }
+  }
```

Comment:
Maybe worth clearing the `events` array after they have been added to the instance.

Reply:
That's a good point, thanks 👍

Reply:
Actually, I made a mistake previously and added events from the freshly added instance (should be the other way around, get the not-yet-flushed events from the previous echo srv instance). Fixed it in [this commit](https://github.com/grafana/grafana/pull/67977/commits/a4dc8a5f0f539f2114403a8ab5afd0767f6b8db9). 

This fix also means that as soon as we override the `singletonInstance`, there is no more reference to the previous echo srv instance, so I think we don't need to clear the events on it (should be garbage collected).